src/spikeinterface/core/waveform_extractor.py

```diff
@@ -1861,8 +1862,20 @@ def __init__(self, waveform_extractor):
         # register
         self.waveform_extractor._loaded_extensions[self.extension_name] = self
 
+    @property
+    def waveform_extractor(self):
+        # Important : to avoid that WaveformExtractor reference a BaseWaveformExtractorExtension
+        # and BaseWaveformExtractorExtension reference a WaveformExtractor
+        # we need a weakref. Otherwise the garbage collecor is not working properly
+        # and so the WaveformExtractor + its recordsing are still alive even after deleting explicitly
+        # the WaveformExtractor which make impossible to delete folder!
+        we = self._waveform_extractor()
+        if we is None:
+            raise ValueError(f"The extension {self.extension_name} has lost its WaveformExtractor")
```

Comment:
```suggestion
        # Important : to avoid the WaveformExtractor referencing a BaseWaveformExtractorExtension
        # and BaseWaveformExtractorExtension referencing a WaveformExtractor
        # we need a weakref. Otherwise the garbage collector is not working properly
        # and so the WaveformExtractor + its recording are still alive even after deleting explicitly
        # the WaveformExtractor which makes it impossible to delete the folder!
        we = self._waveform_extractor()
        if we is None:
            raise ValueError(f"The extension {self.extension_name} has lost its WaveformExtractor")
```